internal/operators/ocs/validations.go

```diff
@@ -0,0 +1,209 @@
+	if err != nil {
+		return err
+	}
+
+	cluster.Operators = string(clusterOperators)
+	return nil
+}
+
+// ValidateOCSRequirements is used to validate min requirements of OCS
+func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
+	var status string
+	hosts := cluster.Hosts
+
+	if int64(len(hosts)) < o.OCSRequiredHosts {
+		status = "Insufficient hosts to deploy OCS. A minimum of 3 hosts is required to deploy OCS. "
+		err := SetOperatorStatus(cluster, status)
+		o.log.Info("Setting Operator Status")
+		if err != nil {
+			o.log.Fatal("Error occured while Marshalling or Unmarshaling Operators ", err)
+			return false
+		}
+		o.log.Info(status)
+		return false
+	}
+	var CPUCount int64 = 0                       //count the total CPUs on the cluster
+	var TotalRAM int64 = 0                       // count the total available RAM on the cluster
+	var DiskCount int64 = 0                      // count the total disks on all the hosts
+	var HostsWithDisks int64 = 0                 // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks                           // to claim for the PV created by LocalVolumeSet
+	if int64(len(hosts)) == o.OCSRequiredHosts { // for only 3 hosts, we need to install OCS in compact mode
+		for _, host := range hosts {
+			err := o.nodeRequirements(host, &CPUCount, &TotalRAM, &DiskCount, &HostsWithDisks)
+			if err != nil {
+				o.log.Fatal("Error occured while calculating Node requirements ", err)
+				return false
+			}
+		}
+
+	} else if len(hosts) == o.OCSMasterWorkerHosts { // not supporting OCS installation for 2 Workers and 1 Master
+		status = "Not supporting OCS Installation for 3 Masters and 2 Workers"
+		o.log.Info("Setting Operator Status")
+		err := SetOperatorStatus(cluster, status)
+		if err != nil {
+			o.log.Fatal("Error occured while Marshalling or Unmarshaling Operators ", err)
+			return false
+		}
+		o.log.Info(status)
+		return false
+	} else if len(hosts) > o.OCSMasterWorkerHosts {
+		for _, host := range hosts { // if the worker nodes >=3 , install OCS on all the worker nodes if they satisfy OCS requirements
+			if host.Role == models.HostRoleWorker {
+				err := o.nodeRequirements(host, &CPUCount, &TotalRAM, &DiskCount, &HostsWithDisks)
+				if err != nil {
+					o.log.Fatal("Error occured while calculating Node requirements ", err)
+					return false
+				}
+			}
+		}
+	}
+	// Hosts with disks must be a multiple of 3
+	o.OCSDisksAvailable = HostsWithDisks / 3
+	canDeployOCS, status := o.validateOCS(o.log, hosts, CPUCount, TotalRAM, DiskCount, HostsWithDisks)
+
+	err := SetOperatorStatus(cluster, status)
+	if err != nil {
+		o.log.Fatal("Error occured while Marshalling or Unmarshaling Operators ", err)
+		return false
+	}
+
+	return canDeployOCS
+}
+
+func (o *ocsValidator) nodeRequirements(host *models.Host, CPUCount *int64, TotalRAM *int64, DiskCount *int64, HostsWithDisks *int64) error {
```

Comment:
Perhaps nodeResources would be a better name as the function is checking the available resources.

Reply:
fixed